File: 1971 Find if Path Exists in Graph/FindIfPathExistsInGraph.cpp

/*
BY RUSHABH GALA
rushabhvg
GitHub :
https://github.com/rushabhvg
StackOverflow :
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        vector<vector<int>> adjList(n);
        queue<int> q;
        vector<bool> seen(n, false);
        int u, v, i, currVer, secondVer;
        for(i=0; i<edges.size(); i++) {
            u=edges[i][0];
            v=edges[i][1];
            adjList[u].push_back(v);
            adjList[v].push_back(u);
        }
        q.push(source);
        seen[source]=true;
        while(!q.empty()) {
            currVer=q.front();
            for(i=0; i<adjList[currVer].size(); i++) {
                secondVer = adjList[currVer][i];
                if (!seen[secondVer]) {
                    q.push(secondVer);
                    seen[secondVer]=true;
                }
            }
            q.pop();
        }
        if (!seen[destination]) {
            return false;
        }
        return true;
    }
};

/*
BY RUSHABH GALA
rushabhvg
GitHub :
https://github.com/rushabhvg
StackOverflow :
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.java
/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        List<List<Integer>> graph = new ArrayList<>();
        for(int i=0; i<n; i++) {
            graph.add(new ArrayList<>());
        }
        for(int[] edge:edges) {
            graph.get(edge[0]).add(edge[1]);
            graph.get(edge[1]).add(edge[0]);
        }
        Queue<Integer> q = new LinkedList<>();
        int[] vis = new int[n];
        vis[source] = 1;
        q.add(source);
        while(!q.isEmpty()) {
            int curr = q.poll();
            if(curr==destination) return true;
            for(int ng : graph.get(curr)) {
                if(vis[ng]==0) {
                    vis[ng] = 1;
                    q.add(ng);
                }
            }
        }
        return false;
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/
